agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt

```diff
@@ -61,6 +63,19 @@ class OfflineAgendaRepository(
                     )
                 }
             }
+        }.onEach {
+            val error = it.getErrorOrNull()
+            error?.let {
+                localAgendaDataSource.saveCreatedAgendaItem(
+                    agendaItemId = agendaItem.id,
+                    userId = sessionStorage.getAuthInfo().userId
+                )
+                syncAgendaItems.syncCreatedAgendaItem(
+                    agendaItemId = agendaItem.id,
+                    agendaItemType = agendaItem.getType()
+                )
+            }
+
```

Comment:
I was confused as to why you could call `onEach()` here, and I saw that your `networkApiCall()` returns a Flow. 

Flows are used when you expect multiple results. You want a single result when you make an API call like this so don’t you think it makes more sense to change the function signature to a suspending one that returns `TaskyResult<T>`?